### PR TITLE
Filter describe-instances so only pending/running instances are selected as etcd peers

### PR DESCRIPTION
**Pull request by stuarthicks**
Specifically, not terminated instances.

### CODE REVIEW

**Comment by tjcorr** [comment]
Weird, I could have sworn we had something like this already in place. Let me test it real quick and I'll get this merged today.

**Comment by tjcorr** [comment]
OK, I went back and read through the original code. I believe we handle this case already with:

etcd_peer_urls=$(aws ec2 describe-instances --region $region --instance-ids $(aws autoscaling describe-auto-scaling-groups --region $region --auto-scaling-group-name $asg_name | jq '.AutoScalingGroups[0].Instances[] | **select(.LifecycleState  == "InService")** | .InstanceId' | xargs) | jq -r ".Reservations[].Instances | map(\"$etcd_client_scheme://\" + .NetworkInterfaces[].PrivateIpAddress + \":$client_port\")[]")

Basically when we grab the nodes from the autoscaling group we are limiting it to those which are active see (http://docs.aws.amazon.com/AutoScaling/latest/APIReference/API_Instance.html). Do we have any cases where this didn't work?

**Comment by stuarthicks** [comment]
Hm, perhaps you're right. I missed that bit. Feel free to close this PR then. I'll re-open if I find a case where this is actually the problem. Sorry.
